landgrab/source/http.py

```diff
@@ -8,20 +11,69 @@ class HTTPSource(BaseSource):
     """
     An input source for HTTP-based network data
     """
-    def __init__(self, uri, method='GET'):
+    def __init__(self, uri, method='GET', paginate=False, results_per_page_param=None,
+                 results_per_page=None, page_index_param=None, max_pages=None, current_page=0,
+                 query_params=None, results_key='results', username=None, password=None):
         self.uri = uri
         self.method = method
+        self.paginate = paginate
+
+        # Vars only necessary if paginate is `True`
```

Comment:
You may want to consider writing it such that the `paginate` option is not a `bool`, but rather a `dict` containing these options. Then there is no conditional here, and it makes it more obvious that these properties are only required if one is trying to paginate results.

Reply:
Cool, change made